app/models/product.rb

```diff
@@ -35,9 +35,12 @@ class Product < ApplicationRecord
   validates_with DiscountValidator
   # validate :discount_price_cannot_be_greater_than_price_value
 
+  before_destroy :ensure_not_referenced_by_any_line_item
   has_many :line_items
   has_many :orders, through: :line_items
-  before_destroy :ensure_not_referenced_by_any_line_item
+
+  after_initialize :ensure_title_has_a_value
```

Comment:
this will be triggered everytime an object is fetched from DB. Not a good approach.
Use `before_validation`